cli/src/commands/operation.rs

```diff
@@ -395,14 +460,541 @@ fn cmd_op_abandon(
     Ok(())
 }
 
+fn cmd_op_show(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &OperationShowArgs,
+) -> Result<(), CommandError> {
+    // TODO: Should we load the repo here?
+    let workspace = command.load_workspace()?;
+    let repo_loader = workspace.repo_loader();
+    let head_op_str = &command.global_args().at_operation;
+    let head_ops = if head_op_str == "@" {
+        // If multiple head ops can't be resolved without merging, let the
+        // current op be empty. Beware that resolve_op_for_load() will eliminate
+        // redundant heads whereas get_current_head_ops() won't.
+        let current_op = op_walk::resolve_op_for_load(repo_loader, head_op_str).ok();
+        if let Some(op) = current_op {
+            vec![op]
+        } else {
+            op_walk::get_current_head_ops(
+                repo_loader.op_store(),
+                repo_loader.op_heads_store().as_ref(),
+            )?
+        }
+    } else {
+        vec![op_walk::resolve_op_for_load(repo_loader, head_op_str)?]
+    };
+    let current_op_id = match &*head_ops {
+        [op] => Some(op.id()),
+        _ => None,
+    };
+    let op = op_walk::resolve_op_for_load(repo_loader, &args.operation)?;
+    let parent_op = merge_operations(repo_loader, command.settings(), op.parents())?;
```

Comment:
Maybe we can simply error out if the operation is a merge?

Since operation can't be merged manually by user, auto-merge result would be identical to the merge operation.

Reply:
Here was [the quick question I asked on Discord](https://discord.com/channels/968932220549103686/969291218347524238/1236281975929704499):

> bnjmnt4n (04/05/2024 19:43)
>
> > What should jj op show with an operation with multiple parents be? Should it merge the parents and attempt to diff with the child? afaik the result of this diff would always be empty, but perhaps that might be different if there are operations which are manually merged?
>
> banana (04/05/2024 20:06)
>
> > the operation log isn't really a general purpose tree, i think it would be ok to tailor the output to the assumption that the merges are ones created by jj merging op heads
>
> Martin von Zweigbergk (04/05/2024 20:14)
> 
> > > What should jj op show with an operation with multiple parents be? Should it merge the parents and attempt to diff with the child? afaik the result of this diff would always be empty, but perhaps that might be different if there are operations which are manually merged?
> >
> > I think so. That would be consistent with what we do for the commit log

I guess it's also possible for a user to manually merge operations using `jj_lib`, which is why it might be useful to diff with the merged operation parents.

Reply:
Yeah, that's technically correct. I was just wondering how we can reduce the amount of the custom code. Maybe we can make `RepoLoader::_resolve_op_heads()` public in some form?